src/crypto/sha256.cpp

```diff
@@ -334,6 +342,40 @@ void Transform(uint32_t* s, const unsigned char* chunk)
     s[5] += f;
     s[6] += g;
     s[7] += h;
+}
+
+/** Define SHA256 hardware */
+#if defined(__linux__)
+#define HWCAP_SHA2  (1<<6)
+#include <sys/auxv.h>
+#elif defined(__WIN64__)
+#include <intrin.h>
+bool isAVX (void) {
+  int cpuinfo[4];
+  __cpuid(cpuinfo, 1);
+  return ((cpuinfo[2] & (1 << 28)) != 0);
+}
+#endif
+
+/** Define a function pointer for Transform */
+void (*transform_ptr) (uint32_t*, const unsigned char*) = &Transform;
+
+/** Initialize the function pointer */
+void inline Initialize_transform_ptr(void)
+{
+// Override the function pointer for ARMV8/AVX2
+#if (defined(USE_ARMV8) || defined(USE_ARMV82))
+    if (getauxval(AT_HWCAP) & HWCAP_SHA2)
+       transform_ptr = &Transform_ARMV8;
+#elif (defined(USE_ARMV82) && defined(__APPLE__))
```

Comment:
This currently does not work on my Apple M1, for 2 reasons:

1. This needs to be swapped places with the above condition, so that order of execution is least inclusive first
2. This needs to compare against both `USE_ARMV8` and `USE_ARMV82` (even though the latter does not work for me yet due to missing `vsha512su0q_u64`) - `FEAT_SHA256` should be available for ARMv8, correct?

Same for `sha1.cpp`